From 936c3484a04cd3469a11d01fecbe7e553b6d0f7f Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:47 -0400
Subject: [scsi] qla2xxx: ISP8044 poll ipmdio bus timeout improvement

Message-id: <1408391817-6389-60-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87943
O-Subject: [PATCH RHEL 7.1 e-stor 59/69] qla2xxx: ISP8044 poll ipmdio bus timeout improvement.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Carnuccio <joe.carnuccio@qlogic.com>

scsi commit id 50a9ff301ed734f2fa95b3c38154454416d934f8

Signed-off-by: Joe Carnuccio <joe.carnuccio@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_nx2.c b/drivers/scsi/qla2xxx/qla_nx2.c
index 26e260e..5e0f4bc 100644
--- a/drivers/scsi/qla2xxx/qla_nx2.c
+++ b/drivers/scsi/qla2xxx/qla_nx2.c
@@ -175,18 +175,17 @@ qla8044_poll_wait_ipmdio_bus_idle(struct scsi_qla_host *vha,
  uint32_t temp;
 
  /* jiffies after 100 msecs */
- timeout = jiffies + (HZ / 1000) * TIMEOUT_100_MS;
+ timeout = jiffies + msecs_to_jiffies(TIMEOUT_100_MS);
  do {
   temp = qla8044_ipmdio_rd_reg(vha, addr1, addr3, mask, addr2);
   if ((temp & 0x1) != 1)
    break;
- } while (!time_after_eq(jiffies, timeout));
-
- if (time_after_eq(jiffies, timeout)) {
-  ql_log(ql_log_warn, vha, 0xb152,
-      "Error in processing mdiobus idle\n");
-  return -1;
- }
+  if (time_after_eq(jiffies, timeout)) {
+   ql_log(ql_log_warn, vha, 0xb152,
+       "Error in processing mdiobus idle\n");
+   return -1;
+  }
+ } while (1);
 
  return 0;
 }
-- 
1.7.1